### DISCUSSION TITLE
Will updateQueryData update transformresponse data automatically?

### DISCUSSION THREAD

**Discussion by slamdunkYYY** [reduxjs/redux-toolkit#3529] [Q&A] [answered]
I have use `transformResponse` in my RTK query, like `transformResponse: (response) => transformData(response)`. And inside the `transformData`, I just create a new object and return back.
```
const transformData = (data) => {
  return {
    ...data,
    allItems: [...data.checkedItems, ...data.uncheckedItems]
  }
}
```
the original data only has `checkedItems` and  `uncheckedItems`.
When I use manual refresh cache, `api.util.updateQueryData`, I found the `allItems` cannot update automatically itself. I have to do another transfer manually too like
```
api.util.updateQueryData("getItemDetails", (details) => {
  const { allItems } = transformData(detail);
  details.allItems = allItems;
});
```
I want to confirm if I have to do the manually update for transformed data in transformResponse when manually cache refresh, or redux toolkit itself can update the transformResponse data when manually cache update?
If I have to use `transformResponse` and `updateQueryData` manually cache update, how can I connect them correctly?

**Comment by EskiMojo14** [comment, marked as answer]
transformResponse is only called when fetching. your updateQueryData call will receive the already transformed data from store, and can operate on it as it chooses.

**Reply by slamdunkYYY** [reply]
The fetching you mean is on the initial api call data, once I got the data from api, then transformResponse transfer the data then data is cached. So transformResponse is only **during the api call**, correct?
But updateQueryData works the **cached** data only, because it's not fetching anymore, it handles the cache, so transformResponse won't be triggered whenI do something on cache.
Is my understanding correct?

**Reply by EskiMojo14** [reply]
correct

**Reply by slamdunkYYY** [reply]
Thank you.
